src/darwin/Framework/CHIP/templates/MTRClusters.zapt

```diff
@@ -34,6 +34,11 @@ NS_ASSUME_NONNULL_BEGIN
                               endpointID:(NSNumber *)endpointID
                                    queue:(dispatch_queue_t)queue NS_DESIGNATED_INITIALIZER {{availability (asUpperCamelCase name preserveAcronyms=true) minimalRelease="First major API revamp"}};
 
+/**
+ * The device this cluster object is associated with.
+ */
+@property (nonatomic, readonly) MTRDevice * device {{availability (asUpperCamelCase name preserveAcronyms=true) minimalRelease="Future"}};
```

Comment:
Can this go on the base class instead of into every concrete cluster?

Reply:
> Can this go on the base class instead of into every concrete cluster?

This is no shared base class that this can go on....  At least until https://github.com/project-chip/connectedhomeip/pull/29241 happens.

Reply:
I'm going to merge this, then rebase #29241 on top of it, since that will be an easier merge, I think, than doing it in the other order: #29241 needs a lot of merge help whether I'm rebasing to master or on top of this PR.